syntax = "proto3";

package weather;

option java_package = "sr.grpc.gen";
option java_outer_classname = "Weather";

enum Relation {
  EQUAL = 0;
  MORE_THAN_EQUAL = 1;
  MORE_THAN = 2;
  LESS_THAN_EQUAL = 3;
  LESS_THAN = 4;
}

enum WeatherDataType {
  TEMPERATURE = 0;
  HUMIDITY = 1;
  ATMOSPHERIC_PRESSURE = 2;
}

message WeatherValue {
  WeatherDataType weatherDataType = 1;
  float value = 2;
}

message WeatherNotifierRequest {
  string cityName = 1;
  WeatherDataType weatherDataType = 2;
  Relation relation = 3;
  float value = 4;
}

message WeatherEvent {
  string detail = 1;
  WeatherDataType weatherDataType = 2;
  repeated WeatherValue currentWeatherValues = 3;
}

service WeatherNotifier {
  rpc subscribeOn(WeatherNotifierRequest) returns (stream WeatherEvent) {}
}

